### PR TITLE
RDK-35695 ControlSettings Plugin Implementation

### PR DESCRIPTION
**Pull request by tamilselvanak**

### CODE REVIEW

File: ControlSettings/ControlSettings.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    void ControlSettings::Deinitialize(PluginHost::IShell* service)
+    {
+        LOGINFO();
+
+	DEVICE_TYPE::getInstance()->Deinitialize();
+    }
+
+    void ControlSettings::InitializeIARM()
+    {
+#if !defined (HDMIIN_4K_ZOOM)
```
Comment (apatel859) [inline ControlSettings/ControlSettings.cpp:72]: why IARM init is linked with HDMIIN_4K_ZOOM

File: ControlSettings/ControlSettings.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+#endif
+    }
+
+    void ControlSettings::DeinitializeIARM()
+    {
+#if !defined (HDMIIN_4K_ZOOM)
+        if (isIARMConnected())
+        {
+            IARM_Result_t res;
+            IARM_CHECK( IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDMI_IN_STATUS) );
```
Comment (apatel859) [inline ControlSettings/ControlSettings.cpp:89]: IARM_Bus_RemoveEventHandler isntead

File: ControlSettings/ControlSettings.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            IARM_CHECK( IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDMI_IN_VIDEO_MODE_UPDATE) );
+            IARM_CHECK( IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDMI_IN_HOTPLUG) );
```
Comment (apatel859) [inline ControlSettings/ControlSettings.cpp:91]: IARM_Bus_RemoveEventHandler

File: ControlSettings/ControlSettings.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    bool ControlSettings::IARMinit() {
+        IARM_Result_t res;
+        bool result = false;
+
+        if (isIARMConnected()) {
+            LOGINFO("ControlSettingsPlugin: IARM already connected");
+            result = true;
+        } else {
+            res = IARM_Bus_Init(PLUGIN_IARM_BUS_NAME);
```
Comment (apatel859) [inline ControlSettings/ControlSettings.cpp:104]: use utils API. Refer any other plugins
ex       if (Utils::IARM::init())

File: ControlSettings/ControlSettingsSTB.cpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        LOGINFO("Exit %s\n",__FUNCTION__);
+    }
+
+    void ControlSettingsSTB::dsHdmiVideoModeEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
+    {
+        LOGINFO("Entry %s\n",__FUNCTION__);
+        LOGINFO("Exit %s\n",__FUNCTION__);
+    }
+    uint32_t ControlSettingsSTB::getVolume(const JsonObject& parameters, JsonObject& response)
+    {
```
Comment (apatel859) [inline ControlSettings/ControlSettingsSTB.cpp:79]: Are getVolume and setVolume dummy API

File: ControlSettings/ControlSettingsTV.cpp [hunk 6]
```diff
@@ -0,0 +1,3920 @@
...
+    ControlSettingsTV::ControlSettingsTV(): PluginHost::JSONRPC()
+					    , m_currentHdmiInResoluton (dsVIDEO_PIXELRES_1920x1080)
+                                            , m_videoZoomMode (tvDisplayMode_NORMAL)
+                                            , m_isDisabledHdmiIn4KZoom (false)
+					    , rfc_caller_id()
+    {
+        LOGINFO("Entry\n");
+        ControlSettingsTV::instance = this;
+	CreateHandler({ 2 });
+        Register("getBacklight", &ControlSettingsTV::getBacklight, this);
```
Comment (apatel859) [inline ControlSettings/ControlSettingsTV.cpp:204]: why Register and registerMethod

File: ControlSettings/ControlSettingsTV.cpp [hunk 7]
```diff
@@ -0,0 +1,3920 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
```
Comment (apatel859) [inline ControlSettings/ControlSettingsTV.cpp:2]: I hope this code is identical from tvsetting thunder plugins

File: ControlSettings/ControlSettingsSTB.cpp (hunk 5 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettingsSTB.cpp:79]: yes..Its Dummy API.

File: ControlSettings/ControlSettings.cpp (hunk 4 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettings.cpp:104]: Changed below API's
1)IARM_Bus_Init -- Utils::IARM::init();
2)isIARMConnected -- Utils::IARM::isConnected();

File: ControlSettings/ControlSettingsTV.cpp (hunk 6 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettingsTV.cpp:204]: In current tvsettings plugin we have different get methods for llama and hisense.
To acheive this, I have followed the same was as "DisplaySettings" implementation.

**For ex :** 
SetBacklight() : We have only one version. so it  registered like below.
registerMethod("setBacklight", &ControlSettingsTV::setBacklight, this);

getBacklight() : We have different get methods for llama and hisense.

Register("getBacklight", &**ControlSettingsTV::getBackligh**t, this);
GetHandler(2)->Register<JsonObject, JsonObject>("getBacklight", &**ControlSettingsTV::getBacklight2**, this);

File: ControlSettings/ControlSettingsTV.cpp (hunk 7 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettingsTV.cpp:2]: Yes. It same as tvsettings-plugins.
I have make this as more generic and moved all the computation logic to HAL layer.

File: ControlSettings/ControlSettings.cpp (hunk 3 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettings.cpp:91]: Thanks. Will Change it accordingly.

File: ControlSettings/ControlSettings.cpp (hunk 1 above)
Comment (tamilselvanak) [reply ControlSettings/ControlSettings.cpp:72]: I have just followed the current tvsettings-plugins.
This flag(HDMIIN_4K_ZOOM) introduced as part of RDK-34823.

**Comment by anand-ky** [comment]
Closing older PRs. Pls bring to latest sprint or topic branch.

**Pull request closed without merging**